### ISSUE TITLE
Traffic splitting deployment based on custom rules

### ISSUE THREAD

**Issue by gsogol** [aws/elastic-beanstalk-roadmap#103]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave "+1" or "me too" comments. They generate extra noise for issue followers and do not help prioritize the request.

**Tell us about your request**
When enabling traffic splitting canaries, currently a certain % of traffic is being routed to determine a successful release. The idea is to be able to 
1) Control that % by letting Beanstalk know which specific users should be routed to the new instances. This can be set by a beanstalk cookie that lets beanstalk know that it should route this particular request. The cookie can be set based on any custom rule determined by the client, may be by some organization, role or any other custom logic. Beanstalk shouldn't care how or why but by simply looking if the cookie is present.
2) a) ability to specify time to keep these new instances in play or b) click a button in aws beanstalk console to force the deployment to continue and finish or a button to switch back to an old instance(s)

**Is this request specific to an Elastic Beanstalk [platform](https://docs.aws.amazon.com/elasticbeanstalk/latest/platforms/platforms-supported.html)?**
If so, which one(s)?

It is not specific to a platform

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
We cannot currently control who gets the new instance and who gets the old instance and when do we determine that it's been successful with more control.

**Are you currently working around this issue?**
How are you currently solving this problem?
No

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by Sekhar-Kutikuppala** [comment]
Hi @gsogol Thank you for the feedback. We will evaluate this. For 2.b)  b) click a button in aws beanstalk console to force the deployment to continue and finish or a button to switch back to an old instance(s)
You can use the Abort deployment option to stop the deployment and switch back to old app version

**Comment by gsogol** [comment]
Yes but also if we say set to keep new instances for 30 min while 30% is being routed to new and I decide after 10min, you know what, everything is fine, no need to wait for 30 set min, want to let beanstalk know manually, all good, force the deployment to finish and route all to new instances. But I understand abort can be used to stop.
